File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
// name -> yearning을 key, value형태로
// photo순회하면서 check
function solution(name, yearning, photo) {
    var answer = [];
    
    const type = {};
    for(let i=0; i<name.length; i++){
        type[name[i]] = yearning[i];
    }
    console.log(type["may"])
    const result = photo.map((val) => 
                          val.reduce((acc, cur) => {
        return acc + (type[cur] || 0)
    },0)
        
    )
    

    return result;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    type = {}
    
    for idx in range(len(name)):
        type[name[idx]] = yearning[idx]
    
    for fir in photo:
        print(fir)
        tmp = 0
        for sec in fir:
            if sec in name:
                tmp += type[str(sec)]
        answer.append(tmp)
    
    return answer